### DISCUSSION TITLE
Using SSH keys for connection in Custom Scripts

### DISCUSSION THREAD

**Discussion by astiks-io** [netbox-community/netbox#15363] [Q&A] [answered]
I have a small script that uses [Junos PyEZ](https://github.com/Juniper/py-junos-eznc) to connect to Juniper devices. This library allows to use SSH keys from file or SSH Agent for connection.
If I run script from the CLI (user is root), the script runs successfully. But if I run it from Web UI, script fails. I suppose that this is due to the script is launched from netbox user.

Is it possible to run scripts via UI from another user?
Or how can I allow netbox user to use ssh agent?

**Comment by candlerb** [comment, marked as answer]
> Is it possible to run scripts via UI from another user?

To do that you would have to create a new queue, with its own netbox-rq worker, running as a different user. But it still wouldn't be able to find your ssh-agent, unless you set the environment variable `SSH_AUTH_SOCK`.  (This is normally present in your shell environment, when you are running commands interactively).

> Or how can I allow netbox user to use ssh agent?

By starting ssh-agent as the parent process of the netbox-rq worker, and then connecting to the agent to load your key and your passphrase (with ssh-add). When invoking ssh-add, you can set `SSH_AUTH_SOCK` explicitly to point to the correct socket location.
